commit 422b1cb0912f88f0509f75b9fee055c8259c13be
Author: Michael Eager <eager@eagercon.com>
Date:   Tue Jan 7 09:15:48 2014 -0800

    2014-01-07  Michael Eager <eager@eagercon.com>
    
       * dwarf2read.c (read_structure_type): Set stub if ICC & length == 0.

diff --git a/gdb/dwarf2read.c b/gdb/dwarf2read.c
index fc4f7cb..2563418 100644
--- a/gdb/dwarf2read.c
+++ b/gdb/dwarf2read.c
@@ -12982,10 +12982,11 @@ read_structure_type (struct die_info *die, struct dwarf2_cu *cu)
       TYPE_LENGTH (type) = 0;
     }
 
-  if (producer_is_icc (cu))
+  if (producer_is_icc (cu) && (TYPE_LENGTH (type) == 0))
     {
       /* ICC does not output the required DW_AT_declaration
 	 on incomplete types, but gives them a size of zero.  */
+      TYPE_STUB (type) = 1;
     }
   else
     TYPE_STUB_SUPPORTED (type) = 1;
